pypeit/spectrographs/vlt_eris.py

```diff
@@ -0,0 +1,360 @@
+
+
+        par['scienceframe']['process']['sigclip'] = 20.0
+        par['scienceframe']['process']['satpix'] ='nothing'
+
+        # Set the default exposure time ranges for the frame typing
+        par['calibrations']['standardframe']['exprng'] = [None, 20]
+        par['calibrations']['arcframe']['exprng'] = [20, None]
+        par['calibrations']['darkframe']['exprng'] = [20, None]
+        par['scienceframe']['exprng'] = [20, None]
+
+        # Sensitivity function parameters
+        par['sensfunc']['algorithm'] = 'IR'
+        par['sensfunc']['polyorder'] = 7
+        par['sensfunc']['IR']['telgridfile'] = 'TellPCA_3000_26000_R10000.fits'
+
+        return par
+
+    def init_meta(self):
+        """
+        Define how metadata are derived from the spectrograph files.
+
+        That is, this associates the PypeIt-specific metadata keywords
+        with the instrument-specific header cards using :attr:`meta`.
+        """
+        self.meta = {}
+        # Required (core)
+        self.meta['ra'] = dict(ext=0, card='RA', required_ftypes=['science', 'standard'])  # Need to convert to : separated
+        self.meta['dec'] = dict(ext=0, card='DEC', required_ftypes=['science', 'standard'])
+        self.meta['target'] = dict(ext=0, card='OBJECT')
+        self.meta['binning'] = dict(ext=0, card=None, default='1,1')
+        self.meta['mjd'] = dict(ext=0, card='MJD-OBS')
+        self.meta['exptime'] = dict(ext=0, card='EXPTIME')
+        self.meta['airmass'] = dict(ext=0, card='HIERARCH ESO TEL AIRM START', required_ftypes=['science', 'standard'])
+        # Extras for config and frametyping
+        # TODO Must change for ERIS
+        self.meta['decker'] = dict(ext=0, card='HIERARCH ESO INS OPTI1 NAME')
+        self.meta['filter1'] = dict(ext=0, card='HIERARCH ESO INS FILT1 NAME')
+        self.meta['dispname'] = dict(ext=0, card='HIERARCH ESO INS GRAT1 NAME')
+
+        self.meta['idname'] = dict(ext=0, card='HIERARCH ESO OCS DET1 IMGNAME')
+        self.meta['instrument'] = dict(ext=0, card='INSTRUME')
+        # self.meta['idname'] = dict(ext=0, card='HIERARCH ESO DPR CATG')
+        # Dithering
+        # TODO Must change for ERIS
+        self.meta['dithoff'] = dict(ext=0, card='HIERARCH ESO OCS CUMOFFS Y ',
+                                   required_ftypes=['science', 'standard'])
+
+    def compound_meta(self, headarr, meta_key):
+        """
+        Methods to generate metadata requiring interpretation of the header
+        data, instead of simply reading the value of a header card.
+
+        Args:
+            headarr (:obj:`list`):
+                List of `astropy.io.fits.Header`_ objects.
+            meta_key (:obj:`str`):
+                Metadata keyword to construct.
+
+        Returns:
+            object: Metadata value read from the header(s).
+        """
+        # TODO Change to match above (init_meta)
+        if meta_key == 'decker':
+            try:  # Science
```

Comment:
In general we prefer to avoid try/except unless really necessary. In this case you could write something like:

```
if meta_key == 'decker':
    if headarr[0].get('HIERARCH ESO INS SLIT NAME') is not None:
        # Science
        decker = headarr[0]['HIERARCH ESO INS SLIT NAME']
    elif headarr[0].get('HIERARCH ESO SEQ SPEC TARG') is not None:
        # Standard!
        decker = headarr[0]['HIERARCH ESO SEQ SPEC TARG']
    else:
        decker = None
return decker
```

Or you can use the same logic with  `if 'HIERARCH ESO INS SLIT NAME' in headarr[0]:`